src/main/java/com/exasol/repository/MavenPomParser.java

```diff
@@ -0,0 +1,163 @@
+        }
+    }
+
+    /**
+     * Parse pom file.
+     * 
+     * @return new instance of {@link MavenPom}
+     */
+    public MavenPom parse() {
+        final String artifactId = parseChildElement("artifactId");
+        final String version = parseChildElement("version");
+        final String deliverableName = parseDeliverableName(artifactId, version);
+        final MavenPom.Builder builder = MavenPom.builder();
+        return builder.version(version).artifactId(artifactId).deliverableName(deliverableName).build();
+    }
+
+    private String parseDeliverableName(final String artifactId, final String version) {
+        final Optional<String> deliverableNameOptional = parseDeliverableName();
+        final String deliverableName;
+        if (deliverableNameOptional.isEmpty()) {
+            deliverableName = artifactId + "-" + version;
+        } else {
+            deliverableName = deliverableNameOptional.get();
+        }
+        return deliverableName;
+    }
+
+    private String parseChildElement(final String elementName) {
+        final Node item = getMandatoryChildNode(elementName);
+        final String element = item.getTextContent().strip();
+        if (element != null && !element.isEmpty()) {
+            return element;
+        } else {
+            throw throwParsingException(elementName);
+        }
+    }
+
+    private Node getMandatoryChildNode(final String elementName) {
+        final Node item = this.parsedPomRoot.getElementsByTagName(elementName).item(0);
+        if (item != null) {
+            return item;
+        } else {
+            throw throwParsingException(elementName);
+        }
+    }
+
+    private IllegalStateException throwParsingException(final String elementName) {
+        return new IllegalStateException(
+                "E-REP-POM-2: Unable to parse pom file because of a missing element: " + elementName);
+    }
+
+    private Optional<String> parseDeliverableName() {
+        final Element build = (Element) this.parsedPomRoot.getElementsByTagName("build").item(0);
+        if (build == null) {
+            return Optional.empty();
+        } else {
+            return parsePlugins(build);
+        }
+    }
+
+    private Optional<String> parsePlugins(final Element build) {
+        final NodeList plugins = build.getElementsByTagName("plugins").item(0).getChildNodes();
+        for (int i = 0; i < plugins.getLength(); i++) {
+            final Node next = plugins.item(i);
+            if (next.getNodeType() == Node.ELEMENT_NODE) {
+                final Element plugin = (Element) next;
+                final Node artifactIdNode = plugin.getElementsByTagName("artifactId").item(0);
+                final String artifactId = artifactIdNode.getTextContent().strip();
+                if (artifactId.equals("maven-assembly-plugin")) {
+                    return parseMavenAssemblyPlugin(plugin);
+                }
```

Comment:
Well, this is hard, because I immediately return a plugin if I found it. If I extract a method here, I will need to add a few if checks and it won't make the code more readable